src/policies/deposits/LimitOrders.sol

```diff
@@ -0,0 +1,516 @@
+            incentiveBudget_,
+            maxPrice_,
+            minFillSize_
+        );
+
+        return orderId;
+    }
+
+    /// @notice Fill a limit order (called by MEV bots)
+    /// @param  orderId_    The ID of the order to fill
+    /// @param  fillAmount_ The amount of USDS to use for the bid
+    function fillOrder(uint256 orderId_, uint256 fillAmount_) external nonReentrant {
+        LimitOrder storage order = orders[orderId_];
+
+        if (!order.active) revert OrderNotActive();
+
+        uint256 remainingDeposit = order.depositBudget - order.depositSpent;
+        if (remainingDeposit == 0) revert OrderFullySpent();
+
+        // Cap fill to remaining deposit budget
+        if (fillAmount_ > remainingDeposit) {
+            fillAmount_ = remainingDeposit;
+        }
+
+        // Check min fill (unless this is the final fill)
+        if (remainingDeposit > order.minFillSize && fillAmount_ < order.minFillSize) {
+            revert FillBelowMinimum();
+        }
+
+        // Calculate proportional incentive
+        // Final fill gets all remaining incentive (avoids rounding dust)
+        uint256 remainingIncentive = order.incentiveBudget - order.incentiveSpent;
+        uint256 incentive;
+        if (fillAmount_ == remainingDeposit) {
+            incentive = remainingIncentive;
+        } else {
+            incentive = (fillAmount_ * order.incentiveBudget) / order.depositBudget;
+        }
+
+        // Check execution price via previewBid
+        uint256 expectedOhmOut = CD_AUCTIONEER.previewBid(order.depositPeriod, fillAmount_);
+        if (expectedOhmOut == 0) revert ZeroOhmOut();
+
+        uint256 effectivePrice = (fillAmount_ * OHM_SCALE) / expectedOhmOut;
+        if (effectivePrice > order.maxPrice) revert PriceAboveMax();
+
+        // Withdraw USDS from sUSDS
+        uint256 usdsNeeded = fillAmount_ + incentive;
+        SUSDS.withdraw(usdsNeeded, address(this), address(this));
+
+        // Update accounting
+        order.depositSpent += fillAmount_;
+        order.incentiveSpent += incentive;
+        totalUsdsOwed -= usdsNeeded;
+
+        // Approve and execute bid
+        USDS.approve(address(CD_AUCTIONEER), fillAmount_);
+
+        (uint256 ohmOut, uint256 positionId, , ) = CD_AUCTIONEER.bid(
+            order.depositPeriod,
+            fillAmount_,
+            expectedOhmOut,
+            true,
+            true
+        );
+
+        // Transfer position NFT to order owner
+        POSITION_NFT.transferFrom(address(this), order.owner, positionId);
+
+        // Transfer receipt tokens to order owner
+        ERC20 receiptToken = receiptTokens[order.depositPeriod];
+        uint256 receiptBalance = receiptToken.balanceOf(address(this));
+        if (receiptBalance > 0) {
+            receiptToken.safeTransfer(order.owner, receiptBalance);
+        }
+
+        // Transfer incentive to filler
+        if (incentive > 0) {
+            USDS.safeTransfer(msg.sender, incentive);
+        }
+
```

Comment:
Is it possible that there is extra USDS? It can be re-wrapped, but I wonder about the accounting.